src/components/calcite-button/calcite-button.scss

```diff
@@ -75,14 +75,14 @@
     margin-right: unset;
   }
 }
-:host([icon-start][dir="rtl"]:not([icon-end])) {
-  .content {
+:host([icon-start]:not([icon-end])) {
+  .calcite--rtl .content {
     margin-left: unset;
```

Comment:
If we are dropping IE support you could just use [logical properties](https://caniuse.com/css-logical-props) here and drop the rtl logic all together.

Reply:
Yeah we should do this at some point. This probably is something we could do for all components